internal/controllers/machineset/machineset_controller.go

```diff
@@ -389,6 +364,43 @@ func (r *Reconciler) reconcile(ctx context.Context, cluster *clusterv1.Cluster,
 	return ctrl.Result{}, nil
 }
 
+// syncMachines updates Machines to propagate in-place mutable fields from the MachineSet.
+// Note: It also cleans up managed fields of all Machines so that Machines that were created/patched before the
+// controller adopted SSA can also work with SSA.  Otherwise fields would be co-owned by our "old" "manager" and
+// "capi-machineset" and then we would not be able to e.g. drop labels and annotations.
+// TODO: update the labels and annotations to the corresponding infra machines and the boostrap configs of the filtered machines.
+func (r *Reconciler) syncMachines(ctx context.Context, machineSet *clusterv1.MachineSet, machines []*clusterv1.Machine) error {
+	log := ctrl.LoggerFrom(ctx)
+	for i := range machines {
+		m := machines[i]
+		// If the machine is already being deleted, we don't need to update it.
+		if !m.DeletionTimestamp.IsZero() {
+			continue
+		}
+
+		// Cleanup managed fields of all Machines.
+		// We do this so that Machines that were created/patched before the controller adopted SSA can also work with SSA.
+		// Otherwise, fields would be co-owned by our "old" "manager" and "capi-machineset" and then we would not be able
+		// to e.g. drop labels and annotations.
+		if err := ssa.CleanUpManagedFieldsForSSAAdoption(ctx, m, machineSetManagerName, r.Client); err != nil {
+			return errors.Wrapf(err, "failed to update machine: failed to adjust the managedFields of the Machine %q", m.Name)
+		}
+
+		// Update Machine to propagate in-place mutable fields from the MachineSet.
+		updatedMachine := r.computeDesiredMachine(machineSet, m)
+		patchOptions := []client.PatchOption{
+			client.ForceOwnership,
+			client.FieldOwner(machineSetManagerName),
+		}
+		if err := r.Client.Patch(ctx, updatedMachine, client.Apply, patchOptions...); err != nil {
+			log.Error(err, "failed to update Machine", "Machine", klog.KObj(updatedMachine))
```

Comment:
is the log + return error pattern intentional or leftover from debugging? I'm assuming the parent function would eventually log the wrapped error

Reply:
It's there intentional.

We weren't sure if to keep it or not but thought it would be beneficial in this case as it's a critical error and the log gives us a log message with the Machine as k/v pair (which we would not get from the log emitted by controller-runtime).

This means that when folks are filtering for the logs of a specific Machine they would also see this log vs. otherwise they would only see it if they filter for the MachineSet